### PR TITLE
Boilerplate

### PR DESCRIPTION
**Pull request by kevinfodness**
* Adds GitHub Actions to run checks on PR creation
* Adds phpcs linting
* Adds PHPUnit support
* Adds .editorconfig
* Adds .gitignore
* Adds composer scripts
* Adds plugin definition
* Adds autoloader
* Adds README
* Does nothing

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+on:
+  pull_request:
+
+jobs:
+  PHPCompatibility:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        php: ['7.4']
```
Comment (jakewrfoster) [inline .github/workflows/linting.yml:12]: should we be prepping this for PHP8?

Comment (kevinfodness) [reply .github/workflows/linting.yml:12]: Yeah, we should, but that's failing right now due to incompatible Composer dependencies. I added an issue about it so we can fix it later: #5 

I had 8.1 and 8.0 in the initial PR but backed them out due to failures for now.

File: .phpcs.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  <!-- Set severity to 1 to see everything that isn't effectively turned off. -->
+  <arg name="severity" value="1"/>
+
+  <!-- Exclude a few directories and autogenerated files. -->
+  <exclude-pattern>vendor/</exclude-pattern>
+
+  <!-- The adapter pattern relies on several files exporting the same class name. -->
+  <rule ref="WordPress.Files.FileName.InvalidClassFileName">
+    <exclude-pattern>adapters/</exclude-pattern>
```
[the remark this answers is not in the document]
Comment (kevinfodness) [reply .phpcs.xml:46]: Yeah, good call, that's how es-admin is structured. es-wp-query has adapters at the top level, but you're right, I should just put everything in lib. I'll make that implementation change in my other branch where I'm wiring up the adapters.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Supported Adapters
+
+* [VIP Enterprise Search](https://docs.wpvip.com/how-tos/vip-search/)
+
+## Usage
+
+Install and activate the plugin to have it interface with an existing installed
+Elasticsearch plugin. This plugin will automatically detect which supported
+Elasticsearch plugin is in use, and will register the appropriate hooks.
```
[the remark this answers is not in the document]
Comment (kevinfodness) [reply README.md:17]: I'm handling that as part of my aggregations work. It'll autoload the adapter for VIP Enterprise Search (the only one I'm building for the moment). We'll need to address it for any other adapters we make (SearchPress, ElasticPress, etc) but that would be part of the issues for creating those adapters.

**Review by jakewrfoster — APPROVED** [review]
a few little questions, but nothing blocking 🙏   🙇

**Review by jakewrfoster — APPROVED** [review]
🐴

**Pull request merged**